criu/cgroup.c

```diff
@@ -716,32 +716,31 @@ static int collect_cgroups(struct list_head *ctls)
 	return 0;
 }
 
-int dump_task_cgroup(struct pstree_item *item, u32 *cg_id, struct parasite_dump_cgroup_args *args)
+int dump_task_cgroup(struct pid *id, u32 *cg_id, struct parasite_dump_cgroup_args *args)
 {
 	int pid;
 	LIST_HEAD(ctls);
 	unsigned int n_ctls = 0;
 	struct cg_set *cs;
 
-	if (item)
-		pid = item->pid->real;
+	if (id)
+		pid = id->real;
 	else
 		pid = getpid();
-
 	pr_info("Dumping cgroups for %d\n", pid);
 	if (parse_task_cgroup(pid, args, &ctls, &n_ctls))
 		return -1;
 
-	cs = get_cg_set(&ctls, n_ctls, item);
+	cs = get_cg_set(&ctls, n_ctls, pid);
```

Comment:
The third argument requested by `get_cg_set` is a bool that is only used once in `if (collect && collect_cgroups(&cs->ctls))` -- since an item that is NULL would have a pid that is also NULL, replacing item with pid will not change the outcome unless future changes are made. 

I'd be happy to elaborate more on this change if there are any other concerns.